x/fungible/keeper/msg_server_update_zrc20_withdraw_fee.go

```diff
@@ -77,8 +74,8 @@ func (k Keeper) UpdateZRC20WithdrawFee(goCtx context.Context, msg *types.MsgUpda
 	)
 	if err != nil {
 		k.Logger(ctx).Error("failed to emit event", "error", err.Error())
-		return nil, sdkerrors.Wrapf(types.ErrEmitEvent, "failed to emit event (%s)", err.Error())
+		return nil, cosmoserrors.Wrapf(types.ErrEmitEvent, "failed to emit event (%s)", err.Error())
 	}
-	commit()
+
```

Comment:
The logic was slightly modified vs old version as the new code commits tmpCtx's effect regardless of event emission failure. Needs to be confirmed.

Reply:
We use `tmpContext` to ensure the `updateProtocolFlatFee` is atomic. The event emission error is a developer error that must never occur.

Reply:
Update: added it back after EmitEvent for consistency as we have this in other messages.

Reply:
Cool man.